pandas/io/excel/_util.py

```diff
@@ -145,7 +159,31 @@ def _range2cols(areas: str) -> list[int]:
     return cols
 
 
-def maybe_convert_usecols(usecols):
+@overload
+def maybe_convert_usecols(usecols: str | list[int]) -> list[int]:
+    ...
+
+
+@overload
+def maybe_convert_usecols(usecols: list[str]) -> list[str]:
+    ...
+
+
+@overload
+def maybe_convert_usecols(
+    usecols: Callable[[Hashable], object]
```

Comment:
I assume it returns the same callable? Could probably use `TypeVar("something", Callable[[Hashable], object])` to return the same function interface.

Reply:
Yep makes sense